File: app/src/main/java/com/gee12/mytetroid/data/ITaskProgress.java
package com.gee12.mytetroid.data;

import com.gee12.mytetroid.logs.TaskStage;
import com.gee12.mytetroid.logs.TetroidLog;

public interface ITaskProgress {

    void nextStage(TetroidLog.Objs obj, TetroidLog.Opers oper, TaskStage.Stages stage);
    boolean nextStage(TetroidLog.Objs obj, TetroidLog.Opers oper, TaskStage.ITaskStageExecutor stageExecutor);
}

File: app/src/main/java/com/gee12/mytetroid/data/ITaskProgress.kt
package com.gee12.mytetroid.data

import com.gee12.mytetroid.common.Either
import com.gee12.mytetroid.logs.LogObj
import com.gee12.mytetroid.logs.LogOper
import com.gee12.mytetroid.logs.TaskStage.Stages

interface ITaskProgress {
    suspend fun nextStage(obj: LogObj, oper: LogOper, stage: Stages)
    suspend fun <L,R> nextStage(obj: LogObj, oper: LogOper, stageExecutor: suspend () -> Either<L,R>): Either<L,R>
}